sklearn/metrics/scorer.py

```diff
@@ -37,10 +38,15 @@ def __init__(self, score_func, sign, kwargs):
         self._kwargs = kwargs
         self._score_func = score_func
         self._sign = sign
+        # XXX After removing the deprecated scorers (v0.20) remove the
+        # XXX deprecation_msg property again and make __call__ abstract again
+        self._deprecation_msg = None
 
-    @abstractmethod
     def __call__(self, estimator, X, y, sample_weight=None):
-        pass
+        if self._deprecation_msg is not None:
+            warnings.warn(self._deprecation_msg,
+                          category=DeprecationWarning,
+                          stacklevel=13)
```

Comment:
For this (or use via `cross_val_score`)
```
scorer = get_scorer('mean_squared_error')
clf.fit(X, y)
scorer(clf, X, y)
```
you need `stacklevel=2` which will point you at `scikit-learn/sklearn/metrics/scorer.py:89`. For the `cross_val_score` you can go higher but then you don't get a warning for the case where you call it directly.

Neither location is particularly informative for the user that didn't expect to get the warning I'd say. either you get that you made this choice somewhere in your code, or you don't get it :-/